include/alpaka/onAcc/warp.hpp

```diff
@@ -0,0 +1,273 @@
+/* Copyright 2025 Sergei Bastrakov, David M. Rogers, Bernhard Manfred Gruber, Aurora Perego, Mehmet Yusufoglu, René
+ * Widera SPDX-License-Identifier: MPL-2.0
+ *
+ * Bridges runtime accelerator instances to the trait-based warp intrinsics so kernels can call them without tags.
+ * Exposes device-safe `alpaka::onAcc::warp::*` wrappers for ballots, shuffles, and lane queries.
+ * Reuses the compile-time warp trait specialisations instead of duplicating backend-specific logic in kernels.
+ * Supplies a uniform warp API across CUDA, HIP, SYCL, and host-emulation accelerators.
+ *
+ * Some example usages:
+ * - consteval `alpaka::getWarpSize(api::Cuda{}, deviceKind::NvidiaGpu{})` for tag-driven compile-time logic.
+ * - device-side `alpaka::onAcc::warp::getSize(acc)` to query the active warp inside a kernel.
+ * - device-side `alpaka::onAcc::warp::shfl(acc, 42, 0u) == 42`
+ */
+
+#pragma once
+
+#include "alpaka/Vec.hpp"
+#include "alpaka/interface.hpp"
+#include "alpaka/onAcc/Acc.hpp"
+#include "alpaka/onAcc/internal/warp.hpp"
+#include "alpaka/tag.hpp"
+
+#include <cstdint>
+
+namespace alpaka::onAcc::warp
+{
+    /** Return the bit-mask of active lanes for the warp associated with the accelerator.
+     *
+     * @return bit mask where the Nth bit is set to 1 if the corresponding thread is participating the call. The return
+     * type can be 64bit or 32bit depending on the API.
+     */
+    template<alpaka::onAcc::concepts::Acc T_Acc>
+    constexpr auto activemask(T_Acc const& acc) -> std::conditional_t<T_Acc::getWarpSize() <= 32u, uint32_t, uint64_t>
+    {
+        using Acc = ALPAKA_TYPEOF(acc);
+        using Api = ALPAKA_TYPEOF(acc[object::api]);
+        return internal::Activemask::Op<Acc, Api>{}(acc, Api{});
+    }
+
+    /** Return the lane index of the current thread within its warp. */
+    constexpr uint32_t getLaneIdx(alpaka::onAcc::concepts::Acc auto const& acc)
+    {
+        using Acc = ALPAKA_TYPEOF(acc);
+        using Api = ALPAKA_TYPEOF(acc[object::api]);
+        return internal::GetLanIdx::Op<Acc, Api>{}(acc, Api{});
```

Comment:
```suggestion
        return internal::GetLaneIdx::Op<Acc, Api>{}(acc, Api{});
```
Otherwise it reads like a local area network idx (Needs to be fixed for all APIs)

Reply:
will be sneaked in with my next pr^^